### ISSUE TITLE
Changing data directory crashes the app

### ISSUE THREAD

**Issue by oleglutsenko** [fabiospampinato/notable#12]
The app is just closed when try to change data directory. The next time you run the app, it uses new location, which is empty.

**Comment by oleglutsenko** [comment]
Just in case, it's about Windows 10.

**Comment by fabiospampinato** [comment]
Thanks for reporting the issue! 

When you first opened the app did it crash after you set the data directory?

**Comment by oleglutsenko** [comment]
Yes, it crashed too.

**Comment by fabiospampinato** [comment]
I'm unable to reproduce this, is anyone would like to help fix this do let me know.

**Comment by dynobo** [comment]
Hi, I'm having the same issue, but in Linux. I tried the [AUR Package](https://aur.archlinux.org/packages/notable-bin/) as well as the 1.1.0 AppImage. 

When I started for the first time, notable asked me to set the directory. After I did, it crashed immediately. Now when I start the app, It seems to work fine, but it only shows the tutorial notes. When I select "Change Data Directory" in the menu, the file dialog opens, but as soon as I select a folder and confirm press "okay", the app crashes again. 

Not sure if it helps, but if I open the AUR version via console, I get a single log message:
```
APPIMAGE env is not defined, current application is not an AppImage
```

When I run the appimage, I get some more messages:
```
[xyz@cioran Downloads]$ ./Notable.1.1.0.AppImage
zenity, kdialog, Xdialog missing. Skipping /tmp/.mount_Notabldvcnn7/AppRun.
Checking for update
Update for version 1.1.0 is not available (latest version: 1.1.0, downgrade is disallowed).
checkForUpdatesAndNotify called, downloadPromise is null
```

Those message appears on any time I open the app. When the app crashs, no log entry is created.

Would love to help with this one, as the editor seems very promising :-)

**Comment by Lythenas** [comment]
I have the same problem using the AUR Package (version 1.4.0). It crashes after selecting the initial location on fist start. And it also crashes when changing locations in the app using "Change Data Directory".

I started the app from the terminal and got the following log. If that help at all.

```
APPIMAGE env is not defined, current application is not an AppImage
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: index.js:216:40
Remote event names: browser-window-created
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: index.js:216:40
Remote event names: browser-window-created
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: index.js:216:40
Remote event names: browser-window-created
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: index.js:216:40
Remote event names: browser-window-created
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: index.js:216:40
Remote event names: browser-window-created
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: index.js:216:40
Remote event names: browser-window-created
```

**Comment by bitbacchus** [comment]
> I'm unable to reproduce this, is anyone would like to help fix this do let me know.

I can reproduce this. However, I'm unfamiliar with Electron & TypeScript. But if you give me some hints where to look, I'm more than happy to help.

**Comment by fabiospampinato** [comment]
@bitbacchus if you've got the time I'd first try to build the project by following [these](https://github.com/notable/notable/blob/master/.github/CONTRIBUTING.md#submit-a-pull-request) instructions, then I think the problematic function might be [this one](https://github.com/notable/notable/blob/961303308f419e763e60500bba34f94486ad6640/src/renderer/containers/main/cwd.ts#L39-L86), which gets called when switching data directory, something is probably going wrong there. I'd recommend putting a few `debugger;` statements here and there so that you can inspect the program at runtime and see if you find anything odd.

**Comment by bitbacchus** [comment]
@fabiospampinato I have some results.

I run Ubuntu 18.04 which comes with node.js 8.10.0. With this version, I got an error when running iconfont and therefore couldn't compile. I installed node.js 12.5.0 and then compiling worked like a charm. And the best: the app didn't crash anymore! I tried the debug mode, the Debian package and AppImage, but couldn't provoke a crash anymore.

So, I'd say its likely some kind of upstream bug and has nothing to do with your code. What do you think?

**Comment by fabiospampinato** [comment]
@bitbacchus thanks for looking into this! 

> And the best: the app didn't crash anymore!

Strange 🤔 the icon font actually only needs to be compiled once, so you should now be able to bundle the app for production using node v8 (maybe you'll need to patch the script in `package.json`, ensuring that the icon font doesn't get rebuild), can you try doing that and seeing if the app crashes?

> So, I'd say its likely some kind of upstream bug and has nothing to do with your code. What do you think?

Yeah it might very well be. It'd still be helpful to understand what makes the app crash though.

**Comment by bitbacchus** [comment]
@fabiospampinato you're right, the icon font is innocent ;-) The trouble compiling it with node v8 is, that it ships with npm v3.5.2 which does not include npx. After installing npx the icons compile nicely. However, with node v8 `npm run dev` throws `Error: Cannot find module 'microevent.ts'`. Not sure if it's worth investigating this further as it just works with node v12? I'd rather explicitly test the node.js version you built the app with. What do you think?

**Comment by fabiospampinato** [comment]
> Error: Cannot find module 'microevent.ts'

I've fixed this by installing "microevent.ts" as a dev dependency, I think `electron-builder` has a bug that causes this package to be deleted or something.

> Not sure if it's worth investigating this further as it just works with node v12?

If the app throws when compiled with node v8 then I can close this issue, because future releases will be compiled with node v12+, otherwise I guess I'll have to keep this open, hoping the underlying issue that causes the problem got fixed somehow and nobody else stumbles upon this.
